libraries/Microsoft.Bot.StreamingExtensions/ContentStream.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using Microsoft.Bot.StreamingExtensions.Payloads;
+
+namespace Microsoft.Bot.StreamingExtensions
+{
+    internal class ContentStream : IContentStream
+    {
+        private readonly ContentStreamAssembler _assembler;
+        private PayloadStream _stream;
+
+        internal ContentStream(Guid id, ContentStreamAssembler assembler)
+        {
+            Id = id;
+            _assembler = assembler ?? throw new ArgumentNullException();
+        }
```

Comment:
_stream = _assembler.GetPayloadStream();

This will let you make _stream readonly, and eliminate the null check below.